### PR TITLE
Add changelog for 4.6.0 (and backdated 4.5.2, 4.5.1)

### PR DESCRIPTION
**Pull request by jakelishman**
This is the proposed changelog for what will be in the new release.  I originally thought this would be a 4.5.3 release, but looking through the features, it really is more like 4.6.0 if we release branched directly off the present-day master, as I had suggested in the emails.  There's plenty of bug-fixes and whatnot included too.

I'm proposing to release most of Sidhant's work from GSoC, Boxi's pulse scheduler and Eric's upgrade to the settings machinery here, along with the support for Numpy 1.20 and a bunch of bug fixes.

Probably the major concern with going to 4.6 is documentation.  We urgently need to merge #117 (Sidhant's docs) and gain documentation for Boxi's scheduler.  I'll also need to go through and check the API-doc sections.  My understanding is that a lot of Eric's settings changes are actually documented already, we just never fulfilled that part of the spec...

If we do decide to go through with this and Sidhant isn't available, I'll rebase his PR, make Boxi's changes and apply it directly.

The prospective release branch exists on the main repository will be `qutip-4.6.X`.

### CODE REVIEW

**Review by quantshah — APPROVED** [review]

**Comment by quantshah** [comment]
Thanks @jakelishman. I agree that 4.6 would be the cleaner approach.

**Review by Ericgig — APPROVED** [review]

**Comment by BoxiLi** [comment]
I can quickly check @sarsid doc and merge it, if he is not available. As far as I see the only conflicts in #117 and #120 is `installation.rst`.  I can also polish all the docs for QIP while adding doc for the scheduler, but I can only do it on Thursday, not tomorrow.

If we are going for 4.6, do you guys think I should also pack up #1356 and get it ready? It is closely related to the scheduler and is already finished, just needs some cleaning. I can get it ready this evening. However, because we want to get the support of NumPy 1.20 ASAP, I'm a bit concerned about if there is anyone willing and have the time to review it?

**Comment by jakelishman** [comment]
I'm not going to have any more time to do anything tonight.  Perhaps we can schedule Friday in as the release day (then if there are any post-launch complications I've got the weekend to fix them), if Boxi you're able to finish off qutip/qutip#1356 and associated documentation before then?  If no-one else has time, I can review the code on a surface level, but I don't really have any experience with `qutip.qip`, so I wouldn't be able to comment much on architecture.

I'm not too concerned with getting #120 merged in ahead of 4.6.0 - it'd be nice, but I don't think it's high enough priority to block the main release.  We can update the docs later, if we need to.

**Comment by BoxiLi** [comment]
> Boxi you're able to finish off qutip/qutip#1356 and associated documentation before then?

Okay, I'll aim for that.

> I'm not too concerned with getting #120 merged in ahead of 4.6.0 - it'd be nice, but I don't think it's high enough priority to block the main release. We can update the docs later, if we need to.

I'm mentioning this because #120 makes the build of docs much easier. Building the current doc is a bit tricky... But you are right that we can update the doc later.

**Comment by jakelishman** [comment]
ah, ok about the build not working.  I have a patch on my local checkout that produces a successful build, so I can just use that to do the release if we can't get #120 merged before Friday

**Review by nwlambert — APPROVED** [review]
thanks for the hard work pushing this through!

**Comment by BoxiLi** [comment]
I have finished all the PRs https://github.com/qutip/qutip/pull/1444, https://github.com/qutip/qutip/pull/1445, https://github.com/qutip/qutip/pull/1446,  https://github.com/qutip/qutip/pull/1447 and the document #127

**Comment by jakelishman** [comment]
Great work Boxi and getting so much done in such a short period of time!

Status of those PRs (ticked means no more action needed):
- [ ] qutip/qutip#1444 (integrate scheduler into compiler)
- [ ] qutip/qutip#1445 (improve noise module)
- [ ] qutip/qutip#1446 (improve pulse plotting)
- [x] qutip/qutip#1447 (fix LaTeX circuit images for measurement)
- [ ] qutip/qutip-doc#127 (noise, processor and scheduler documentation)

File: changelog.rst [hunk 1]
```diff
@@ -1,12 +1,132 @@
 .. QuTiP
-   Copyright (C) 2011-2020, Paul D. Nation, Robert J. Johansson, Alexander Pitchford, Chris Granade, Arne Grimsmo, Nathan Shammah, Shahnawax Ahmed, Eric Giguère
+   Copyright (C) 2011-2020, Paul D. Nation, Robert J. Johansson, Alexander Pitchford, Chris Granade, Arne Grimsmo, Nathan Shammah, Shahnawaz Ahmed, Jake Lishman, and Eric Giguère
```
Comment (hodgestar) [inline changelog.rst:2]: :smiley_cat:

File: changelog.rst [hunk 2]
```diff
@@ -1,12 +1,132 @@
...
+This is the first release to have full binary wheel releases on pip; you can now do ``pip install qutip`` on almost any machine to get a correct version of the package without needing any compilers set up.
+The support for Numpy 1.20 that was first added in QuTiP 4.5.3 is present in this version as well, and the same build considerations mentioned there apply here too.
+If building using the now-supported PEP 517 mechanisms (e.g. ``python -mbuild /path/to/qutip``), all build dependencies will be correctly satisfied.
+
+Improvements
+------------
+- **MAJOR** Add saving, loading and resetting functionality to ``qutip.settings`` for easy re-configuration. (by **Eric Giguère**)
+- **MAJOR** Add a quantum gate scheduler in ``qutip.qip.scheduler``, to help parallelise the operations of quantum gates.  This supports two scheduling modes: as late as possible, and as soon as possible. (by **Boxi Li**)
+- **MAJOR** Improved qubit circuit simulators, including OpenQASM support and performance optimisations. (by **Sidhant Saraogi**)
+- **MAJOR** Add tools quantum measurements and their statistics. (by **Simon Cross** and **Sidhant Saraogi**)
```
Comment (hodgestar) [inline changelog.rst:24]:
```suggestion
- **MAJOR** Add tools for quantum measurements and their statistics. (by **Simon Cross** and **Sidhant Saraogi**)
```

**Review by hodgestar — APPROVED** [review]
Found one small missing word, but otherwise looks good to me (with the caveat that I wasn't closely involved during 2020, so I wouldn't know if anything was missed).

File: changelog.rst [hunk 3]
```diff
@@ -1,12 +1,132 @@
...
+- Add support for Numpy 1.20.  QuTiP should be compiled against a version of Numpy ``>= 1.16.6`` and ``< 1.20`` (note: does _not_ include 1.20 itself), but such an installation is compatible with any modern version of Numpy.  Source installations from ``pip`` understand this constraint.
+- Improve the error message when circuit plotting fails. (by **Boxi Li**)
+- Add support for parsing M1 Mac hardware information. (by **Xiaoliang Wu**)
+- Support decomposition of ``X``, ``Y`` and ``Z`` gates in circuits. (by **Boxi Li**)
```
Comment (BoxiLi) [inline changelog.rst:28]:
```suggestion
- Add more single-qubit gates and controlled gates. (by **Mateo Laguna** and **Martín Sande Costa**)
- Support decomposition of ``X``, ``Y`` and ``Z`` gates in circuits. (by **Boxi Li**)
- Refactor ``QubitCircuit.resolve_gate()`` (by **Mateo Laguna**)
```
See https://github.com/qutip/qutip/pull/1209, https://github.com/qutip/qutip/pull/1224 and https://github.com/qutip/qutip/pull/1226

File: changelog.rst [hunk 4]
```diff
@@ -1,12 +1,132 @@
...
+- Fix ``ptrace`` when all subspaces are being kept, or the subspaces are passed in order. (by **Jake Lishman**)
+- Fix sorting bug in ``Bloch3d.add_points()``. (by **pschindler**)
+- Fix invalid string literals in docstrings and some unclosed files. (by **Élie Gouzien**)
+- Fix Hermicity tests for matrices with values that are within the tolerance of 0. (by **Jake Lishman**)
+- Fix the trace norm being incorrectly reported as 0 for small matrices. (by **Jake Lishman**)
+- Fix issues with ``dnorm`` when using CVXPy 1.1 with sparse matrices. (by **Felipe Bivort Haiek**)
+- Fix segfaults in ``mesolve`` when passed a bad initial ``Qobj`` as the state. (by **Jake Lishman**)
+- Fix sparse matrix construction in PIQS when using Scipy 1.6.1. (by **Drew Parsons**)
+- Fix ``zspmv_openmp.cpp`` missing from the pip sdist. (by **Christoph Gohlke**)
+- Fix correlation functions throwing away imaginary components. (by **Asier Galicia Martinez**)
```
Comment (BoxiLi) [inline changelog.rst:45]: https://github.com/qutip/qutip/pull/1269 and https://github.com/qutip/qutip/pull/1194
```suggestion
- Fix correlation functions throwing away imaginary components. (by **Asier Galicia Martinez**)
- Fix ``QubitCircuit.add_circuit()`` for SWAP gate. (by **Canoming**)
- Fix the brocken LaTeX image conversion. (by **Jake Lishman**)
```

**Review by BoxiLi — COMMENTED** [review]
I checked the PR history and there might be a few related to qip missing. Would be good to add them as an encouragement.

(There is even one PR of yours :)

**Comment by jakelishman** [comment]
Thanks both!  Boxi, not sure how I missed so many PRs from earlier, so thanks for catching them.  It must have happened when I originally made this PR - when I updated it last night, I only checked merge commits that happened after February.

File: changelog.rst [hunk 5]
```diff
@@ -1,12 +1,136 @@
...
+- **MAJOR** Add a quantum gate scheduler in ``qutip.qip.scheduler``, to help parallelise the operations of quantum gates.  This supports two scheduling modes: as late as possible, and as soon as possible. (by **Boxi Li**)
+- **MAJOR** Improved qubit circuit simulators, including OpenQASM support and performance optimisations. (by **Sidhant Saraogi**)
+- **MAJOR** Add tools for quantum measurements and their statistics. (by **Simon Cross** and **Sidhant Saraogi**)
+- Add support for Numpy 1.20.  QuTiP should be compiled against a version of Numpy ``>= 1.16.6`` and ``< 1.20`` (note: does _not_ include 1.20 itself), but such an installation is compatible with any modern version of Numpy.  Source installations from ``pip`` understand this constraint.
+- Improve the error message when circuit plotting fails. (by **Boxi Li**)
+- Add support for parsing M1 Mac hardware information. (by **Xiaoliang Wu**)
+- Add more single-qubit gates and controlled gates. (by **Mateo Laguna** and **Martín Sande Costa**)
+- Support decomposition of ``X``, ``Y`` and ``Z`` gates in circuits. (by **Boxi Li**)
+- Refactor ``QubitCircuit.resolve_gate()`` (by **Mateo Laguna**)
```
Comment (BoxiLi) [inline changelog.rst:30]: Oh no, I made a mistake with the name. Could you rebase again? Very sorry
```suggestion
- Refactor ``QubitCircuit.resolve_gate()`` (by **Martín Sande Costa**)
```

**Review by BoxiLi — APPROVED** [review]

**Pull request merged**